File: remote-config/client/src/main/java/br/com/leonardoferreira/poc/Application.java
package br.com.leonardoferreira.poc;

import lombok.RequiredArgsConstructor;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@SpringBootApplication
@ConfigurationPropertiesScan
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

}

@ConfigurationProperties(prefix = "custom")
record MyProp(String simple) {
}

@RestController
@RequiredArgsConstructor
class MyController {

    private final MyProp myProp;

    @GetMapping("/prop")
    String prop() {
        return myProp.simple();
    }

}

File: kafka/spring-cloud-stream-integrated-spring-kafka/spring-kafka/src/main/kotlin/br/com/leonardoferreira/Application.kt
package br.com.leonardoferreira

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.boot.runApplication
import org.springframework.scheduling.annotation.EnableScheduling

@EnableScheduling
@SpringBootApplication
@ConfigurationPropertiesScan
class Application

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}
